systems/forecast_combine.py

```diff
@@ -340,7 +340,12 @@ def _get_trading_rule_list_with_fixed_weights(self, instrument_code: str) -> lis
                 # nested dict
                 rules = config.forecast_weights[instrument_code].keys()
             else:
-                # assume it's a non nested dict, eg weights same across instruments
+                # seems it's a non nested dict (weights same across instruments), but let's check 
+                # that just in case it IS nested dict but instrument weight is missing
+                for val in config.forecast_weights.values():
+                    if isinstance(val, dict):
+                        # so it is a nested dict..
+                        raise Exception("Missing forecast weight for instrument ",instrument_code)
```

Comment:
I really hate this

Reply:
Ah, sorry for the spaghetti code. I tried to figure out a more elegant solution to this but I didn't see a better way.

At least for now it gives a proper error message (it was quite perplexing to see all rules being named as instrument codes and to find out that the culprit was that I had mistyped an instrument code in forecast weights list... :)